client/src/connection_cache.rs

```diff
@@ -275,6 +275,9 @@ mod tests {
         )
         .unwrap();
 
+        let response_recv_endpoint = response_recv_endpoints
```

Comment:
Not strictly related to this PR because pre-change the test is doing the same,
but: why is it using one of the server endpoints? Why isn't it creating a new
endpoint? Makes the test confusing - I had to triple check that it's fine to
just pick one from the now multi-endpoints.

Maybe add a comment that explicitly says that it doesn't really matter which
endpoint is used?